sdcm/sct_config.py

```diff
@@ -2533,27 +2896,27 @@ def update_argus_with_version(self, scylla_version: str, package_name: str):
         except Exception as exc:  # pylint: disable=broad-except
             self.log.exception("Failed to save target Scylla version in Argus", exc_info=exc)
 
-    def dict(self):
-        out = deepcopy(self)
-
-        # handle pydantic object, and convert them back to dicts
-        # TODO: automate the process if we gonna keep using them more, or replace the whole configuration with pydantic/dataclasses
-        if kafka_connectors := self.get('kafka_connectors'):
-            out['kafka_connectors'] = [connector.dict(by_alias=True, exclude_none=True)
-                                       for connector in kafka_connectors]
-        if mgmt_restore_params := self.get("mgmt_restore_params"):
-            out["mgmt_restore_params"] = mgmt_restore_params.dict(by_alias=True, exclude_none=True)
-        if mgmt_agent_backup_config := self.get("mgmt_agent_backup_config"):
-            out["mgmt_agent_backup_config"] = mgmt_agent_backup_config.dict(by_alias=True, exclude_none=True)
-        return out
+    # def dict(self):
```

Comment:
this is still something trick left to handle, the configuration which need to be translated to pydantic

the code as it's own' probably doesn't support it correctly now, i.e. it won't translate from string to pedantic model, and not from dict/yaml